pkg/accumulator/accumulator.go

```diff
@@ -0,0 +1,256 @@
+		containerID: containerID,
+	}
+}
+
+func (acc *Accumulator) createNewMap(event *evData.EventData, index int) {
+	slice := make([]evData.EventData, 0)
+	m := make(map[string][]evData.EventData)
+	m[event.GetEventContainerID()] = slice
+	acc.accumulatorData[index] = m
+	acc.firstMapKeysOfAccumulatorData[index] = event.GetEventContainerID()
+}
+
+func (acc *Accumulator) getFirstTimestamp() (time.Time, error) {
+	if len(acc.accumulatorData) < 1 {
+		return time.Time{}, fmt.Errorf("getFirstTimestamp failed the slice accumulatorData has no members")
+	}
+	if len(acc.firstMapKeysOfAccumulatorData) < 1 {
+		return time.Time{}, fmt.Errorf("getFirstTimestamp failed the slice firstMapKeysOfAccumulatorData has no members")
+	}
+	if len(acc.accumulatorData[0][acc.firstMapKeysOfAccumulatorData[0]]) < 1 {
+		return time.Time{}, fmt.Errorf("getFirstTimestamp failed the slice acc.accumulatorData[0][acc.firstMapKeysOfAccumulatorData[0] has no members")
+	}
+	return acc.accumulatorData[0][acc.firstMapKeysOfAccumulatorData[0]][0].GetEventTimestamp(), nil
+}
+
+func (acc *Accumulator) findIndexByTimestampWhenAccumulatorDataIsFull(event *evData.EventData) int {
+	index := 0
+	minTimestamp, err := acc.getFirstTimestamp()
+	if err != nil {
+		logger.L().Warning("findIndexByTimestampWhenAccumulatorDataIsFull fail to find the place to insert the event, fail with error", helpers.Error(err))
+		return -1
+	}
+	for i := range acc.accumulatorData {
+		if i == 0 {
+			continue
+		}
+		if acc.accumulatorData[i][acc.firstMapKeysOfAccumulatorData[i]][0].GetEventTimestamp().Before(minTimestamp) {
+			minTimestamp = acc.accumulatorData[i][acc.firstMapKeysOfAccumulatorData[i]][0].GetEventTimestamp()
+			index = i
+		}
+	}
+	acc.createNewMap(event, index)
+	return index
+}
+
+func (acc *Accumulator) findIndexByTimestamp(event *evData.EventData) int {
+	for i := range acc.accumulatorData {
+		if len(acc.accumulatorData[i]) == 0 {
+			acc.createNewMap(event, i)
+			return i
+		}
+		firstKey := acc.firstMapKeysOfAccumulatorData[i]
+		if event.GetEventTimestamp().Sub((acc.accumulatorData[i])[firstKey][0].GetEventTimestamp()) < time.Second {
+			return i
+		}
+	}
+	return acc.findIndexByTimestampWhenAccumulatorDataIsFull(event)
+}
+
+func (acc *Accumulator) removeAllStreamedContainers(event *evData.EventData) {
+	acc.containersData.registerMutex.Lock()
+	if len(acc.containersData.accumulatorDataPerContainer) > 0 {
+		for contID := range acc.containersData.accumulatorDataPerContainer {
+			acc.containersData.accumulatorDataPerContainer[contID] <- *event
+		}
+	}
+	acc.containersData.registerMutex.Unlock()
+}
+
+func (acc *Accumulator) addEventToCacheAccumulator(event *evData.EventData, index int) {
+	acc.syncReaderWriterAccumulatorData.Lock()
+	a := acc.accumulatorData[index]
+	a[event.GetEventContainerID()] = append(a[event.GetEventContainerID()], *event)
+	acc.accumulatorData[index][event.GetEventContainerID()] = append(acc.accumulatorData[index][event.GetEventContainerID()], *event)
```

Comment:
Do you mean: 
```suggestion
	acc.accumulatorData[index][event.GetEventContainerID()] = a[event.GetEventContainerID()]
```

What am I missing here?